File: SecondYear/semestrul4/MPP/examen_practic/Rezolvari/boat/boat/networking/src/main/java/org/networking/utils/RpcConcurrentServer.java
package org.networking.utils;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.networking.rpcprotocol.ProjectClientRpcWorker;
import org.services.ProjectServices;

import java.net.Socket;

public class RpcConcurrentServer extends AbsConcurrentServer {
    private ProjectServices server;
    private static final Logger logger = LogManager.getLogger(RpcConcurrentServer.class);


    public RpcConcurrentServer(int port, ProjectServices server) {
        super(port);
        this.server = server;
        logger.info("RpcConcurrentServer created on port {}", port);
    }

    @Override
    protected Thread createWorker(Socket client) {
        ProjectClientRpcWorker worker = new ProjectClientRpcWorker(server, client);
        Thread thread=new Thread(worker);
        return thread;
    }

}

File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Networking/utils/RpcConcurrentServer.cs

using System.Net.Sockets;
using log4net;
using Services;
using Networking.rpcprotocol;

namespace Networking.utils
{
    public class RpcConcurrentServer : AbsConcurrentServer
    {
        private IProjectServices server;
        private static readonly ILog log = LogManager.GetLogger(typeof(AbsConcurrentServer));

        public RpcConcurrentServer(int port, IProjectServices server) : base(port)
        {
            this.server = server;
            log.Error("RpcConcurrentServer created on port {port}" + port);
        }

        protected override Thread CreateWorker(TcpClient client)
        {
            ProjectClientRpcWorker worker = new ProjectClientRpcWorker(server, client);
            return new Thread(worker.Run); 
        }
    }
}
